graphdatascience/arrow_client/v2/gds_arrow_client.py

```diff
@@ -0,0 +1,509 @@
+from __future__ import annotations
+
+import json
+import logging
+from types import TracebackType
+from typing import Any, Callable, Type
+
+import pandas
+import pyarrow
+from pyarrow import RecordBatch, flight
+
+from graphdatascience.arrow_client.arrow_endpoint_version import ArrowEndpointVersion
+from graphdatascience.arrow_client.authenticated_flight_client import AuthenticatedArrowClient, ConnectionInfo
+
+from ...procedure_surface.api.default_values import ALL_TYPES
+from ...procedure_surface.utils.config_converter import ConfigConverter
+from .api_types import JobStatus
+from .job_client import JobClient
+
+
+class GdsArrowClient:
```

Comment:
i have been thinking a bit about the different versioning.

how about we call this one `GdsArrowClientV2` (rename the v1 similar).
and have a common interface `GdsArrowClient` for the common api? 
If you want to use version specific methods you can cast it on usage

Reply:
ok now i see you only want to return the job id in the API here.
that makes it more difficult indeed

Reply:
I started with extracting a shared interface but then realized that the APIs are too different. 
I kept the name the same after looking at Pydantic. There they keep the same class names and have them in separate packages.

Reply:
pytest is just complaining about the same test file name i think.
but okay can keep it as is